core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -23,7 +24,7 @@ class WorkflowCompiler(
   def compileLogicalPlan(
       logicalPlanPojo: LogicalPlanPojo,
       executionStateStore: ExecutionStateStore
-  ): LogicalPlan = {
+  ): Option[LogicalPlan] = {
```

Comment:
Can we throw errors out and handle them at one place? This method is better to return `LogicalPlan` and only fails with exceptions.

Reply:
done